parsimonious/expressions.py

```diff
@@ -218,15 +217,25 @@ class Literal(Expression):
     Use these if you can; they're the fastest.
 
     """
-    __slots__ = ['literal']
+    __slots__ = ['literal', 'identity_tuple']
 
     def __init__(self, literal, name=''):
         super(Literal, self).__init__(name)
         self.literal = literal
+        self.identity_tuple = (name, literal)
+
+    def __hash__(self):
+        return hash(self.identity_tuple)
+
+    def __eq__(self, other):
```

Comment:
ISTM that we could factor this up into Expression itself by having it check `self.__class__`. Thoughts?